app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/uploadsupportingdocs/UploadSupportingEvidenceController.scala

```diff
@@ -0,0 +1,1012 @@
+                  fillingOutReturn.agentReferenceNumber
+                )
+          _ <- EitherT(
+                updateSession(sessionStore, request)(
+                  _.copy(journeyStatus = Some(fillingOutReturn.copy(draftReturn = newDraftReturn)))
+                )
+              )
+        } yield ()
+
+        result.fold({ e =>
+          logger.warn("Could not update session", e)
+          errorHandler.errorResult()
+        }, _ => Redirect(controllers.returns.routes.TaskListController.taskList()))
+      }
+    }
+
+  @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements", "org.wartremover.warts.Any"))
+  private def checkYourAnswersHandler(
+    answers: UploadSupportingEvidenceAnswers,
+    fillingOutReturn: FillingOutReturn,
+    draftReturn: DraftReturn
+  )(implicit request: RequestWithSessionData[_]): Future[Result] =
+    answers match {
+      case IncompleteUploadSupportingEvidenceAnswers(None, _) =>
+        Redirect(routes.UploadSupportingEvidenceController.doYouWantToUploadSupportingDocuments())
+      case IncompleteUploadSupportingEvidenceAnswers(Some(doYouWantToUploadSupportingEvidence), supportingEvidences) =>
+        if (doYouWantToUploadSupportingEvidence) {
+          supportingEvidences match {
+            case Nil => Redirect(routes.UploadSupportingEvidenceController.uploadSupportingEvidence())
+            case ::(_, _) =>
+              val uploadedSupportingEvidences = for {
+                documents <- upscanService.getAll(DraftReturnId(draftReturn.id.toString))
+              } yield documents
+              uploadedSupportingEvidences.fold(
+                e => InternalServerError(s"could not get all uploaded supporting evidence: $e"),
+                s => {
+                  val expiredSupportingEvidence = findExpiredSupportingEvidence(s)
+                  if (expiredSupportingEvidence.nonEmpty) {
+                    Ok(expiredSupportingEvidencePage(expiredSupportingEvidence))
+                  } else {
+                    Ok(
+                      checkYourAnswersPage(
+                        CompleteUploadSupportingEvidenceAnswers(
+                          doYouWantToUploadSupportingEvidence,
+                          supportingEvidences
+                        )
+                      )
+                    )
+                  }
+                }
+              )
+          }
+        } else {
+          val uploadedSupportingDocuments = for {
+            documents <- upscanService.getAll(DraftReturnId(draftReturn.id.toString))
+          } yield documents
+
+          uploadedSupportingDocuments.fold(
+            e => InternalServerError(s"could not get all uploaded supporting evidence: $e"),
+            se => {
+              val expiredSupportingEvidence = findExpiredSupportingEvidence(se)
+              if (expiredSupportingEvidence.nonEmpty) {
+                Ok(expiredSupportingEvidencePage(expiredSupportingEvidence))
```

Comment:
the user has chosen not to upload any evidence - why would we look for expired supporting evidence? There shouldn't be any

Reply:
there should be - we don't have a TTL on the collection. The reason is we need to replay info back to the user when they come back to service after 7 days. 
In the next ticket I will work on to do the dms integration, I am going to make a call to the collection to delete all uploaded evidences for a particular draft id if the draft was successfully submitted to ETMP.